### PR TITLE
fix(kuma-cp): use pagination store for secret store

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

SecretStore wasn't using a pagination store so there was a risk that resources might not be sorted. Changed SecretStore to use pagination store.

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/secrets/k8s/store.go [hunk 1]
```diff
@@ -302,9 +301,6 @@ func (c *SimpleConverter) ToCoreList(in *kube_core.SecretList, out core_model.Re
 			}
 			secOut.Items[i] = r
 		}
-		sort.SliceStable(secOut.Items, func(i, j int) bool {
```
Comment (slonka) [inline pkg/plugins/secrets/k8s/store.go:305]: to get this right, the previous store could have partial response here and this is how it was not sorted?

**Review by slonka — COMMENTED** [review]
does a pagination store enforce sorting?

**Comment by lukidzi** [comment]
> does a pagination store enforce sorting?

Yes, https://github.com/kumahq/kuma/blob/master/pkg/core/resources/store/pagination_store.go#L82

File: pkg/plugins/secrets/k8s/store.go (hunk 1 above)
Comment (lukidzi) [reply pkg/plugins/secrets/k8s/store.go:305]: Not sure if that could be the case. I think we may have overlooked sorting in this situation due to the delegation model, where tasks are delegated from one entity to another. The secret store wasn't positioned behind the pagination store, which ensures that results are sorted.

**Review by slonka — APPROVED** [review]

**Pull request merged**